src/main/java/com/microsoftopentechnologies/windowsazurestorage/AzureStorageBuilder.java

```diff
@@ -310,53 +291,45 @@ public synchronized void perform(Run<?, ?> run, FilePath workspace, Launcher lau
             } else {
                 listener.getLogger().println(Messages.AzureStorageBuilder_files_downloaded_count(filesDownloaded));
             }
-
         } catch (IOException | InterruptedException | WAStorageException e) {
             e.printStackTrace(listener.error(Messages.AzureStorageBuilder_download_err(storageAccountInfo.getStorageAccName())));
             run.setResult(Result.UNSTABLE);
         }
     }
 
-    private int downloadArtifacts(final Run<?, ?> source, final AzureStorageBuilderContext context) {
-        int filesDownloaded = 0;
-        try {
-            final AzureBlobAction action = source.getAction(AzureBlobAction.class);
-            List<AzureBlob> azureBlobs = action.getIndividualBlobs();
-            if (action.getZipArchiveBlob() != null && includeArchiveZips) {
-                azureBlobs.addAll(Arrays.asList(action.getZipArchiveBlob()));
-            }
-
-            filesDownloaded = WAStorageClient.downloadArtifacts(azureBlobs, context);
-
-        } catch (WAStorageException e) {
-            context.getRun().setResult(Result.UNSTABLE);
-        }
+    private StoragePluginService getDownloadService(final DownloadServiceData data) {
+        if (isDownloadFromContainer())
+            return new DownloadFromContainerService(data);
+        return new DownloadFromBuildService(data);
```

Comment:
No, directly from file share is not supported.  Can we add to backlog another task for this feature? to avoid this PR be expanded again.